src/sysext/sysext.c

```diff
@@ -262,9 +277,38 @@ static int unmerge_hierarchy(
                 if (r == 0)
                         break;
 
+                r = read_one_line_file(work_dir_info_file, &base64_work_dir);
+                if (r < 0) {
+                        if (r != -ENOENT)
+                                return log_error_errno(r, "Failed to read '%s': %m", work_dir_info_file);
+                }
+
+                if (base64_work_dir) {
+                        r = unbase64mem(base64_work_dir, (void**)&work_dir, NULL);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to base64 decode work directory path: %m");
+                }
+
+                r = path_is_mount_point(dot_dir, NULL, 0);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to determine whether '%s' is a mount point: %m", dot_dir);
+                // TODO: Should this check be stricter? Like checking if "p" is a mountpoint over itself with
+                // a read-only option set?
```

Comment:
why have the check at all? why not just do umount() and handle gracefully if we get EINVAL because path is not a mountpont?

Reply:
Sounds good, done. I was also thinking about using umount2 directly instead of umount_verbose to avoid spamming journal with messages about failing to unmount a directory. But for now demoted those to debug messages and the error will be printed when we get something else than EINVAL.